### DISCUSSION TITLE
Nothing's changed even after adding an SSH key to GitHub

### DISCUSSION THREAD

**Discussion by dense-set** [community/community#57590] [New to GitHub] [answered]
#### [This is a copy-paste of the [post](https://stackoverflow.com/q/76436858/11953295) I recently posted on StackOverflow, which is not attracting much attention. Hence thought of asking here.]

I followed the [documentation on GitHub][1] to set up an SSH key for automatic authentication. The proof that I have set it up correctly is the test mentioned in the doc:

    $ ssh -T git@github.com
    Hi USERNAME! You've successfully authenticated, but GitHub does not provide shell access.

I am able to `clone` repos from GitHub alright to my system. However, I am not able to `push` anything (to reps where I can write). I get the following:

    $ git push origin main         
    Username for 'https://github.com': USERNAME
    Password for 'https://USERNAME@github.com': 
    remote: Support for password authentication was removed on August 13, 2021.
    remote: Please see https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories#cloning-with-https-urls for information on currently recommended modes of authentication.
    fatal: Authentication failed for 'https://github.com/dense-set/test_repo.git/'

I have tried entering my passphrase for the SSH key as well as my GitHub password, and the same `fatal` error shows.

**Question:** How can I successfully `push`?

  [1]: https://docs.github.com/en/authentication/connecting-to-github-with-ssh/about-ssh

**Comment by TuralJafar** [comment]
qwerty

**Reply by dense-set** [reply]
eh?

**Comment by airtower-luna** [comment, marked as answer]
If `git push` asks for HTTPS credentials, that means the repository is still configured with an HTTPS URL for the remote, so it's not using SSH at all. You'll need to update the URL for `origin` to use SSH instead (you can find the right URL behind the green "Code" button):

```
git remote set-url origin git@github.com:dense-set/test_repo.git
```

Double check that the URL is correct with `git remote -v`.

**Reply by dense-set** [reply]
This worked, thanks!
